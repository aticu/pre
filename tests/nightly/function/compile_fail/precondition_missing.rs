use pre::pre;

#[pre("is bar")]
#[pre("is baz")]
fn foo() {}

#[pre]
fn main() {
    #[assure("is bar", reason = "foo is bar")]
    foo()
}
