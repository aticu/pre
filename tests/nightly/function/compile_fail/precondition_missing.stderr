error[E0308]: mismatched types
 --> $DIR/precondition_missing.rs:9:5
  |
9 |     #[assure("is bar", reason = "foo is bar")]
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected a tuple with 2 elements, found one with 1 element
  |
  = note: expected struct `std::marker::PhantomData<(pre::CustomCondition<"is bar">, pre::CustomCondition<"is baz">)>`
             found struct `std::marker::PhantomData<(pre::CustomCondition<"is bar">,)>`
